misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -162,12 +170,24 @@ internal class RealTransacter private constructor(
     val TRANSACTER_SPAN_TAG = "hibernate-transacter"
   }
 
-  internal class RealSession(
-    val session: org.hibernate.Session,
-    val config: DataSourceConfig
-  ) : Session {
+  internal class ReadOnlySession(
+    session: org.hibernate.Session,
+    config: DataSourceConfig
+  ) : BaseSession(session, config) {
     override val hibernateSession = session
+    init {
+      hibernateSession.isDefaultReadOnly = true
+      hibernateSession.hibernateFlushMode = FlushMode.MANUAL
+    }
+    override fun <T : DbEntity<T>> save(entity: T): Id<T> {
+      throw IllegalStateException("Saving isn't permitted in a read only session.")
+    }
+  }
 
+  internal class WritableSession(
```

Comment:
lets chat about this . . .  I think readonly/readwrite is a great idea, but not a good fit for our type hierarchy

Reply:
Switched. We talked about layers of code that need to be modified if something is readonly/readwite, and List.remove.